### PR TITLE
fix!: Use %APPDATA% instead of ~/.twilio-cli on Windows

### PR DESCRIPTION
**Pull request by sryze**
Use the `%APPDATA%` directory on Windows instead of the home directory. 

On Windows all applications keep configuration and caches inside the AppData directory. Command-line tools should not be an exception. Besides, it does not make much sense to create a "dot" directory on Windows because such directories do not bear any special meaning, unlike on *nix systems, and for example are not hidden by the file explorer. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli-core/blob/main/CONTRIBUTING.md) and my PR follows them **<-- this URL is not working BTW**
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: src/services/env.js [hunk 1]
```diff
@@ -4,8 +4,12 @@ const path = require('path');
 const { CLI_NAME } = require('./config');
 
 const configureEnv = () => {
-  const home = process.env.HOME || process.env.USERPROFILE || os.homedir();
-  const twilioDir = path.join(home, `.${CLI_NAME}`);
+  const twilioDir = (() => {
+    if (process.platform === 'win32' && process.env.APPDATA) {
+      return path.join(process.env.APPDATA, CLI_NAME);
+    }
+    return path.join(process.env.HOME || os.homedir(), `.${CLI_NAME}`);
```
Comment (thinkingserious) [inline src/services/env.js:11]: Shouldn't we still have `process.env.USERPROFILE` here?

File: test/services/env.test.js [hunk 2]
```diff
@@ -50,17 +51,17 @@ describe('services', () => {
...
 
-        expect(process.env.TWILIO_CACHE_DIR).to.equal(path.join('user-profile', '.twilio-cli'));
+        expect(process.env.TWILIO_CACHE_DIR).to.equal(path.join('appdata', 'twilio-cli'));
       });
 
-      test.it('should fallback to the OS home dir if no home or user profile found', () => {
+      test.it('should fallback to the OS home dir if no home or AppData found', () => {
         delete process.env.HOME;
-        delete process.env.USERPROFILE;
+        delete process.env.APPDATA;
```
Comment (thinkingserious) [inline test/services/env.test.js:64]: Why do we need to remove this line? `delete process.env.USERPROFILE;`

File: test/services/env.test.js [hunk 3]
```diff
@@ -6,14 +6,15 @@ const { expect, test } = require('@twilio/cli-test');
...
+const DEFAULT_DIR =
+  process.platform === 'win32' ? path.join('appdata', 'twilio-cli') : path.join('home', '.twilio-cli');
 
 describe('services', () => {
   describe('env', () => {
     describe('configureEnv', () => {
       beforeEach(() => {
         process.env.HOME = 'home';
-        process.env.USERPROFILE = 'user-profile';
+        process.env.APPDATA = 'appdata';
```
Comment (thinkingserious) [inline test/services/env.test.js:17]: I think we still need `process.env.USERPROFILE = 'user-profile';`

**Review by thinkingserious — COMMENTED** [review]
Hello @sryze!

Thank you for the PR, we greatly appreciate it!

Looks like we also have a test update needed based on the failure here: https://travis-ci.com/github/twilio/twilio-cli-core/jobs/485365714#L523

File: src/services/env.js (hunk 1 above)
Comment (sryze) [reply src/services/env.js:11]: Yes, you are right, I'll add it back

Comment (thinkingserious) [reply src/services/env.js:11]: Thank you!

**Comment by JenniferMah** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**